### PR TITLE
Feature/test market get assets

### PR DESCRIPTION
**Pull request by Lex-23**

### CODE REVIEW

File: cryptostock/market/models.py [hunk 1]
```diff
@@ -1,27 +1,36 @@
...
         """ Return all allowed assets from market """
 
     @abc.abstractmethod
-    def get_asset(self, name) -> Dict:
+    def get_asset(self, name: str) -> Asset:
         """ Return asset by name """
 
     @abc.abstractmethod
-    def buy(self, name, count: int) -> Dict[Dict, int]:
+    def buy(self, name: str, count: Count) -> Dict[Asset, Count]:
```
Comment (vadim-denisov) [inline cryptostock/market/models.py:33]: Typing is wrong.

File: cryptostock/market/tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+try:
+    from account.tests.conftest import *  # noqa
+except ImportError:
+    pass
```
Comment (vadim-denisov) [inline cryptostock/market/tests/conftest.py:4]: Why that?

Comment (vadim-denisov) [reply cryptostock/market/tests/conftest.py:4]: try-exept I mean.

File: cryptostock/market/tests/factory.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+from factory.django import DjangoModelFactory
+from market.models import Market
+
+assets_list = [
```
Comment (vadim-denisov) [inline cryptostock/market/tests/factory.py:4]: Should be fixture. If some test modify `assets_list` not, we will have modified object in next test. Fixture will also return new object.

File: cryptostock/market/models.py [hunk 4]
```diff
@@ -1,27 +1,39 @@
...
+class Asset(TypedDict):
+    name: str
+    description: str
+    price: str
+
+
+class BuyResponse(TypedDict):
+    asset: Asset
+    count: int
+    status: str
```
Comment (vadim-denisov) [inline cryptostock/market/models.py:19]: You don't have status in line 60

File: cryptostock/market/tests/conftest.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        {"name": "ADA", "description": "Cardano", "price": "1.4383"},
+        {"name": "USDC", "description": "USDCoin", "price": "1.0001"},
+        {"name": "XRP", "description": "XRP", "price": "0.832515"},
+        {"name": "DOT1", "description": "Polkadot", "price": "28.54"},
+        {"name": "LUNA1", "description": "Terra", "price": "69.42"},
+        {"name": "HEX", "description": "HEX", "price": "0.136504"},
+        {"name": "DOGE", "description": "Dogecoin", "price": "0.178727"},
+        {"name": "AVAX", "description": "Avalanche", "price": "96.68"},
+    ]
+    return assets_list
```
Comment (vadim-denisov) [inline cryptostock/market/tests/conftest.py:22]:
```suggestion
def assets_list():
    return [
        {"name": "BTC", "description": "Bitcoin", "price": "50996.99"},
        {"name": "ETH", "description": "Ethereum", "price": "4353.60"},
        {"name": "BNB", "description": "BinanceCoin", "price": "586.37"},
        {"name": "USDT", "description": "Tether", "price": "1.0012"},
        {"name": "SOL1", "description": "Solana", "price": "199.38"},
        {"name": "ADA", "description": "Cardano", "price": "1.4383"},
        {"name": "USDC", "description": "USDCoin", "price": "1.0001"},
        {"name": "XRP", "description": "XRP", "price": "0.832515"},
        {"name": "DOT1", "description": "Polkadot", "price": "28.54"},
        {"name": "LUNA1", "description": "Terra", "price": "69.42"},
        {"name": "HEX", "description": "HEX", "price": "0.136504"},
        {"name": "DOGE", "description": "Dogecoin", "price": "0.178727"},
        {"name": "AVAX", "description": "Avalanche", "price": "96.68"},
    ]
```

File: cryptostock/market/tests/test_get_assets.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+from django.db import connection
+from django.test.utils import CaptureQueriesContext
+from market.models import YahooMarket
+from market.tests.factory import MarketFactory
+
+
+def test_get_assets_from_market_name(auth_broker, get_assets_list):
+    name = "Yahoo"
+    YahooMarket.get_assets = MagicMock(return_value=get_assets_list)
+    MarketFactory(name=name)
```
Comment (vadim-denisov) [inline cryptostock/market/tests/test_get_assets.py:12]: Why do we need factory?

Comment (vadim-denisov) [reply cryptostock/market/tests/test_get_assets.py:12]: I think factory should present in DB. We can add fixture with this model and sync it to DB, before we deploy app to dev/prod.

File: cryptostock/market/models.py (hunk 1 above)
Comment (Lex-23) [reply cryptostock/market/models.py]: fixed

File: cryptostock/market/tests/conftest.py (hunk 2 above)
Comment (Lex-23) [reply cryptostock/market/tests/conftest.py]: fixed

File: cryptostock/market/models.py (hunk 4 above)
Comment (Lex-23) [reply cryptostock/market/models.py:19]: got it

File: cryptostock/market/tests/conftest.py (hunk 5 above)
Comment (Lex-23) [reply cryptostock/market/tests/conftest.py:22]: fixed

File: cryptostock/market/tests/factory.py (hunk 3 above)
Comment (Lex-23) [reply cryptostock/market/tests/factory.py]: done

File: cryptostock/market/tests/conftest.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import pytest
+from account.tests.conftest import *  # noqa
+from market.models import Market
+
+
+@pytest.fixture
+def get_assets_list():
```
Comment (vadim-denisov) [inline cryptostock/market/tests/conftest.py:7]:
```suggestion
def assets_list():
```

File: cryptostock/market/tests/conftest.py [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+        {"name": "DOT1", "description": "Polkadot", "price": "28.54"},
+        {"name": "LUNA1", "description": "Terra", "price": "69.42"},
+        {"name": "HEX", "description": "HEX", "price": "0.136504"},
+        {"name": "DOGE", "description": "Dogecoin", "price": "0.178727"},
+        {"name": "AVAX", "description": "Avalanche", "price": "96.68"},
+    ]
+
+
+@pytest.fixture
+def get_market(db):
```
Comment (vadim-denisov) [inline cryptostock/market/tests/conftest.py:26]:
```suggestion
def yahoo_market(db):
```

Comment (Lex-23) [reply cryptostock/market/tests/conftest.py]: this fixture do not use already, removed

File: cryptostock/market/tests/conftest.py (hunk 7 above)
Comment (Lex-23) [reply cryptostock/market/tests/conftest.py]: done

**Comment by Lex-23** [comment]
[Write MarketClient](https://trello.com/c/AEeD8oPC/10-write-marketclient)

**Review by vadim-denisov — APPROVED** [review]

**Pull request merged**